packages/backend/src/ee/models/AiAgentModel.ts

```diff
@@ -1674,4 +1682,286 @@ export class AiAgentModel {
+                AiArtifactVersionsTableName,
+            )
+                .select(
+                    trx.raw(
+                        'COALESCE(MAX(version_number), 0) + 1 as next_version',
+                    ),
+                )
+                .where('ai_artifact_uuid', data.artifactUuid)
+                .first<{ next_version: number }>();
+
+            const nextVersion = result?.next_version ?? 1;
+
+            // Insert new version
+            const [version] = await trx<AiArtifactVersionsTable>(
+                AiArtifactVersionsTableName,
+            )
+                .insert({
+                    ai_artifact_uuid: data.artifactUuid,
+                    ai_prompt_uuid: data.promptUuid,
+                    version_number: nextVersion,
+                    title: data.title ?? null,
+                    description: data.description ?? null,
+                    viz_config_output: data.vizConfig,
+                })
+                .returning('*');
+
+            if (!version) {
+                throw new Error('Failed to create artifact version');
+            }
+
+            // Get artifact info to return complete object
+            const artifact = await trx<AiArtifactsTable>(AiArtifactsTableName)
+                .select('*')
+                .where('ai_artifact_uuid', data.artifactUuid)
+                .first();
+
+            if (!artifact) {
+                throw new NotFoundError('Artifact not found');
+            }
+
+            return {
+                artifactUuid: artifact.ai_artifact_uuid,
+                threadUuid: artifact.ai_thread_uuid,
+                artifactType: artifact.artifact_type,
+                savedQueryUuid: artifact.saved_query_uuid,
+                createdAt: artifact.created_at,
+                versionNumber: version.version_number,
+                title: version.title,
+                description: version.description,
+                vizConfigOutput: version.viz_config_output,
+                promptUuid: version.ai_prompt_uuid,
+                versionCreatedAt: version.created_at,
+            };
+        });
+    }
+
+    async createOrUpdateArtifact(data: {
+        threadUuid: string;
+        promptUuid: string;
+        artifactType: 'chart';
+        title?: string;
+        description?: string;
+        vizConfig: Record<string, unknown>;
+    }): Promise<AiArtifact> {
+        const existingArtifact = await this.findArtifactByThreadUuid(
+            data.threadUuid,
+        );
+
+        if (existingArtifact) {
+            return this.createArtifactVersion({
+                artifactUuid: existingArtifact.artifactUuid,
+                promptUuid: data.promptUuid,
+                title: data.title,
+                description: data.description,
+                vizConfig: data.vizConfig,
+            });
+        }
+
+        return this.createArtifact(data);
+    }
+
+    async getArtifact(artifactUuid: string): Promise<AiArtifact> {
```

Comment:
could we update this to optionally include version number? if no version is selected, defaults to latest (and update it to `getArtifactVersion`)